### ISSUE TITLE
Setting (xah-fly-keys 1) doesn't work in --daemon mode in startup

### ISSUE THREAD

**Issue by wofwofwof** [xahlee/xah-fly-keys#103]
When I use my config with emacs everything works fine, when I use it with emac --daemon mode the xah-fly-key mode is activated but ignored until I 

- open a file
- deactivate and activate it again

After this everything works fine. I use version xah-fly-keys-20200707.1310.

At the moment I have no clue whats the reason for this behaviour. On xah-fly-keys-20190223.716 that configuration works. Can somebody just try if its my configuration or a general problem with xah-fly-mode?

Thanks for that fantastic mode

cheers

alex

**Comment by wofwofwof** [comment]
ok, found my problem, just cleaned up my old dependencies in the packages and now iit works. No problem with xah-fly-mode. Sorry for the noise

**Comment by wofwofwof** [comment]
ok, found my problem, just cleaned up my old dependencies in the packages and now iit works. No problem with xah-fly-mode. Sorry for the noise

**Comment by wofwofwof** [comment]
lol, reopened again

its still open after the cleanup, but I've got confused because the keys with C- mapping are working. And there are lots of triggers that get one back into the fly-spell mode, some examples are listed:

- press Meta-x
- press keys for interactive search
- press keys for file open ...

is it possible that some part of xah now needs an interactive environment and that on daemon mode that doesen't work?

**Comment by TRSx80** [comment]
It sound to me like possible config conflict?  What do you think?

Have you tried to reproduce the problem with a minimal config (i.e., `emacs --no-init-file` or `-q`)?

**Comment by wofwofwof** [comment]
Hello TRSx80,

thanks for your support. I've tried to make a minimal config where the problem is still there. This is my .emacs.d/init.el:

(require 'package)
(add-to-list 'package-archives '("melpa" . "https://melpa.org/packages/"))
(package-initialize)


(require 'xah-fly-keys)
(xah-fly-keys-set-layout "qwertz")
(xah-fly-keys 1)


Thank you very much

wof

**Comment by CelestialNebula** [comment]
I don't know why it happens but this fixed it for me.

Taken from:
https://github.com/daviwil/emacs-from-scratch/blob/master/show-notes/Emacs-Tips-08.org#configuring-the-ui-for-new-frames

```
(defun my/server-fix-up()
  "Make sure 'xah-fly-keys' is starting in command-mode.

https://github.com/xahlee/xah-fly-keys/issues/103
https://github.com/daviwil/emacs-from-scratch/blob/master/show-notes/Emacs-Tips-08.org#configuring-the-ui-for-new-frames"
  (xah-fly-keys-set-layout "dvorak")
  (xah-fly-keys t))

(if (daemonp)
    (add-hook 'server-after-make-frame-hook 'my/server-fix-up))
```

**Comment by wofwofwof** [comment]
CelestialNebula,
thanks a lot, that fixed the problem for me, too.

**Comment by maisiliym** [comment]
```elisp

**Comment by eexexiexit** [comment]
If i use this fix, in the daemon mode, some calendar bindings seem to have been damaged (in insert mode), for instance (after M-x calendar) and switch to insert-mode:

i d  -> diary-insert-entry
i b  -> diary-insert-block-entry

Does anybody has an idea of what is going on here?

**Comment by CelestialNebula** [comment]
What do you mean damaged? Those appear to be the default keys.

**Comment by xahlee** [comment]
maybe open a new issue and reference to this one.
i'll look into.
meanwhile, i don't recommend running daemon. it creates a lot complications. is confusing in concept,  also a lot technical complexities. also more complexities if you are on windows.

instead, setup ur emacs so that u can open multiple instances without emacs ever bugging u to ask questions during launch (e.g. last emacs crashed). so you can always work in mulitple intances in parallel, launch new anytime. if u code emacs, u have to resart emacs often, sometimes multiple times a day, anyway.
